lib/hanami/application/settings/dotenv_store.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "dry/core/constants"
+
+module Hanami
+  class Application
+    class Settings
+      # Default application settings store.
+      #
+      # Uses dotenv (if available) to load .env files and then loads settings
+      # from ENV.
+      #
```

Comment:
We should document which files are loaded (according to the current env), and what's their priority.

Reply:
Done!